### ISSUE TITLE
Need few More Visualization Features

### ISSUE THREAD

**Issue by igs3000** [usnistgov/NetSimulyzer#31]
I think, Visualization of features such as  Packet Transmission Event, Packet Drop Event are not yet implemented in NetSimulyzer.

Visualizing Packet Transmission Events as an expanding sphere will a pleasant experience (that AquaSim3D and Omnet++ are providing). 

Even though there is no support for Visualizing underwater network scenarios under ns-3, I just did 'proof of concept simulation' for UWSN using NetSimulyzer. 
[Simulating And Visualizing 3D Mobile Underwater Sensor Network (UWSN) Under ns-3](https://www.projectguideline.com/simulating-and-visualizing-3d-mobile-underwater-sensor-network-uwsn-under-ns-3/)

If you compare the same 'proof of concept simulation'  of UWSN that I did using Omnet++, you can understand the difference in the visualization features that we are missing in NetSimulyzer.
[Simulating and Visualizing 3D Underwater Wireless Sensor Network (UWSN) Under Omnet++](https://www.projectguideline.com/simulating-and-visualizing-3d-underwater-wireless-sensor-network-uwsn-under-omnet/)

If I am missing something in this regard, please let me know it.

Charles Pandian

**Comment by bpe2** [comment]
Hey Charles,

Thanks for the feedback. I'd definitely like to visualize packet transmission somehow. The expanding sphere thing sounds like a decent way to do that, and visually that's not a difficult effect to implement. I'll ask around, and if others are interested, I'll get on it.

It seems like there are plenty of other smaller things the NetSimulyzer could do as well, like the motion trails shown in your Omnet++ example. I'll break them out into some other issues as I see them.

Evan

**Comment by igs3000** [comment]
Hello Evan,
I saw your presentation on wns3-2022.
You demonstrated the feature of “motion trails” in that video. 
Is it available in the public version of Netsimulyzer?

**Comment by bpe2** [comment]
Yep, it's off by default, though. You can find it in File->Settings, in the "Graphics" section it's "Show Motion Trails".

(In this screenshot, it's turned on)
![image](https://user-images.githubusercontent.com/54951512/181294947-0ccc530a-dadf-4cb2-a77c-c4e05d91aa9a.png)

**Comment by igs3000** [comment]
Thank you.
Understood.

Is there any other way to enable/disable from the ns-3 simulation script itself? (so that, it can be scheduled as per the user's wish during the simulation itself?)

**Comment by bpe2** [comment]
Right now, no.

I guess I always just saw this as up to user preference, and not necessarily a per-simulation thing, Since it can be turned on and off easily (but changing the trail length does need a restart), so I'm not sure if I would add it as something configurable in the simulation

**Comment by igs3000** [comment]
Expected to enable/disable from the simulation because, sometimes, we may only need to visualize the motion trails of some selected nodes. Anyway, you may ignore this for the time being. Because there are so many important and attractive features (like automating transmission sphere), we may need to implement.

**Comment by bpe2** [comment]
I never thought of turning it on or off on a particular `Node`, but I guess that does make sense. I'll mention this, and see if others would be interested in this sort of control.
